### ISSUE TITLE
Method group in the constructor of the System.Lazy class does not work properly

### ISSUE THREAD

**Issue by traf72** [dotnet/runtime#46310]
# Issue Title

If I provide a method group in the constructor of the System.Lazy class it doesn't work properly.

# General

Actual for .NET 5. I created a simple console app to illustrate the problem [ConsoleApp1.zip](https://github.com/dotnet/core/files/5724308/ConsoleApp1.zip).

Code Example:

```c#
using System;
using System.Reflection;

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            var lazy_1 = new Lazy<Assembly>(() => Assembly.GetExecutingAssembly());
            Console.WriteLine(lazy_1.Value);
            
            var lazy_2 = new Lazy<Assembly>(Assembly.GetExecutingAssembly);
            Console.WriteLine(lazy_2.Value);
        }
    }
}
```
In this code the first console statement prints:

`ConsoleApp1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null`

But the second console statement prints:

`System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`

But in fact the code is the same. How is it possible?

**Comment by MichaeIDietrich** [comment]
OK, I ran your code example and you are right. I am also surprised that both lines produce the same result in .NET Core 3.1, which I did not expect though (same for .NET Framework). For some reason I expected that it was always the behavior which we can see in .NET 5.0. 
Nevertheless this makes it a breaking change.

Be careful with code suggestions from IDEs. Usually those suggestions are indeed helpful and clean up the code, but in rare cases they can lead to unwanted behavior, as in your example here. E.g. another example is transforming a loop into a LINQ expression, which might lead to worse performance if it is executed _very frequently_, so never apply those suggestions thoughtlessly.
